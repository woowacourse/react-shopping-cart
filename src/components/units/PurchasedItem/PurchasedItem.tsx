import React from 'react';
import defaultImageURL from 'assets/images/brave.png';
import * as T from 'types';
import Styled from './PurchasedItem.styles';
import Button from '../../shared/Button/Button';

type PurchasedItemProps = {
  item: T.CartItem;
  onClick: (product: T.Product) => void;
};

const PurchasedItem = (props: PurchasedItemProps) => {
  const { item, onClick } = props;
  const { quantity, product } = item;
  const { name, price, image } = product;

  const totalPrice = price * quantity;

  const handleClick = () => {
    onClick(product);
  };

  return (
    <Styled.Root>
      <Styled.Image src={image ?? defaultImageURL} alt={name} />
      <Styled.Info>
        <Styled.Title>{name}</Styled.Title>
        <Styled.Detail>
          {totalPrice.toLocaleString('ko-KR')}원 / 수량 : {quantity}개
        </Styled.Detail>
      </Styled.Info>
      <Styled.ButtonWrapper>
        <Button size={T.ButtonSize.REGULAR} text="장바구니" onClick={handleClick} />
      </Styled.ButtonWrapper>
    </Styled.Root>
  );
};

export default PurchasedItem;
