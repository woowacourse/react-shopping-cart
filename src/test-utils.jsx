import React from 'react';
import { Provider } from 'react-redux';
import { Global, ThemeProvider } from '@emotion/react';
import { SnackbarProvider } from 'notistack';
import { render } from '@testing-library/react';
import { store } from './modules/store';
import { globalStyle, theme } from './App.styles';

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <SnackbarProvider
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'center',
        }}
        maxSnack={3}
      >
        <Global styles={globalStyle} />
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </SnackbarProvider>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
