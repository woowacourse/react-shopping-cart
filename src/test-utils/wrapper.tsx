import { MemoryRouter } from "react-router-dom";
import { ToastProvider } from "../contexts/ToastContext";
import { CartProvider } from "../contexts/CartContext";
import { CouponProvider } from "../contexts/CouponContext";

const wrapper = ({ children }: React.PropsWithChildren) => (
  <MemoryRouter>
    <ToastProvider>
      <CartProvider>
        <CouponProvider>{children}</CouponProvider>
      </CartProvider>
    </ToastProvider>
  </MemoryRouter>
);

export default wrapper;
