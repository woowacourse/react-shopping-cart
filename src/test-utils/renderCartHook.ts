import { renderHook } from "@testing-library/react";
import useCart from "../hooks/useCart";
import useCartCalculations from "../hooks/useCartCalculations";
import wrapper from "./wrapper";

export const renderCartHook = () =>
  renderHook(
    () => {
      const cart = useCart();
      const calculations = useCartCalculations();
      return { ...cart, ...calculations };
    },
    { wrapper }
  );
