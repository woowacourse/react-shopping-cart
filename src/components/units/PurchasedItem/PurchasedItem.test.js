import React from 'react';
import { render } from '../../../test-utils';
import PurchasedItem from './PurchasedItem';

describe('PurchasedItem 테스트', () => {
  test('matches snapshot', () => {
    const item = {
      productId: 1,
      name: '재밌는 상품',
      price: 50000,
      quantity: 5,
    };

    const utils = render(<PurchasedItem item={item} />);

    expect(utils.container).toMatchSnapshot();
  });
});
