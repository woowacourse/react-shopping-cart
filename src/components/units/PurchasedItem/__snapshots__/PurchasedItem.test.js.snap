// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurchasedItem 테스트 matches snapshot 1`] = `
<div>
  <li
    class="css-mmzmx8-Root e1ycuyj35"
  >
    <img
      alt="item-image"
      class="css-lbmuvq-Image e1ycuyj34"
      src="brave.png"
    />
    <div
      class="css-ecoywl-Info e1ycuyj33"
    >
      <div
        class="css-1nsby4x-Title e1ycuyj32"
      >
        재밌는 상품
      </div>
      <div
        class="css-13fe54v-Detail e1ycuyj31"
      >
        250,000
        원 / 수량 : 
        5
        개
      </div>
    </div>
    <div
      class="css-4uuqdl-ButtonWrapper e1ycuyj30"
    >
      <button
        class="css-fsw4cv-Root e6rpj8n0"
      >
        장바구니
      </button>
    </div>
  </li>
</div>
`;
